186939310 -  **[MultiDoc]** Clue Google Analytics Metrics

dennisrcao opened the pull request:
Fulfills this ticket 
[**[Platform]** Instrument CLUE Google Analytics Metrics
](https://www.pivotaltracker.com/n/projects/2441242/stories/186939310)

The following parameters should be logged and can be seen on Google Analytics (copied from ticket)
 Page, problem, the number of accesses,  time it took to load,  version of browser, and OS. 
 IP address cannot be collected due to legal reasons. 

Deployed link: 
https://collaborative-learning.concord.org/branch/instrument-clue-performance-metrics/?appMode=demo&demoName=CLUE&fakeClass=1&fakeUser=student:1&problem=0.1
_____________________________________________________________


LocalHost: 
http://localhost:8080/?appMode=demo&demoName=CLUE&fakeClass=1&fakeUser=student:1&problem=0.1

scytacki reviewed (changes requested):
I left a review in slack. Here it is for completeness:

It looks to me like what you are doing with the unit and problem will work. We probably will need to setup custom metrics or dimensions for them so we can break down the audience by them.  Perhaps you already did that, if not, this page seems to describe how to do that (but I haven't read it): https://support.google.com/analytics/answer/14240153?visit_id=638429384510460419-3220538971&rd=1

For the loadingStartTime this should be useful.  I'd recommend adding another event like this called `loadingComplete` (or something like that) when  `logLoadingAndDocumentMeasurements` is called. You don't need to include all of the measurements just the time.  So I think this would mean putting the following code at the beginning of logLoadingAndDocumentMeasurements:
```
gtag("event", "loadingComplete", {
  "event_category": "Performance",
  "event_label": "Custom Metric - loadingComplete",
  "value": getTimestamp()
});
```

You'll have to tell typescript about gtag so it doesn't complain. This can be done by:
running `npm install --save-dev @types/gtag.js`
then in tsconfig.json change:
`"types": ["w3c-web-serial"]` to `"types": ["w3c-web-serial", "gtag.js"]`

You might need to restart VSCode typescript after this by hitting `CMD+Shift+P` then typing `restart ts` and select the first option.

dennisrcao commented:
@scytacki Just getting back to this.
So yes I did already setup 2 custom dimensions for `problem` and `unit` as two parameters for `page_view` in our GA > CLUE
<img width="1908" alt="image" src="https://github.com/concord-consortium/collaborative-learning/assets/521934/4b650cc8-1a2e-4193-ae9a-30b469465b6c">

dennisrcao commented:
@scytacki 
Got it. I removed them from Custom dimensions and added them to Custom metrics (screenshot below) 
<img width="1542" alt="image" src="https://github.com/concord-consortium/collaborative-learning/assets/521934/615cb4af-3a33-490a-bd68-279f5fa545f0">
